lib/reek/context/code_context.rb

```diff
@@ -68,8 +68,9 @@ def initialize(_parent, exp)
       # @param type [Symbol] the type of the nodes we are looking for, e.g. :defs.
       # @yield block that is executed for every node.
       #
-      def local_nodes(type, &blk)
-        each_node(type, [:casgn, :class, :module], &blk)
+      def local_nodes(type, ignored = [], &blk)
+        ignored += [:casgn, :class, :module]
```

Comment:
But it doesn't. The original reference will still be intact since `+=` will mutate a locally scoped array:

>>[1] pry(main)> foo = [:class]
=> [:class]
[2] pry(main)> def bar(arg); arg += [:module]; end
=> :bar
[3] pry(main)> bar foo
=> [:class, :module]
[4] pry(main)> foo
=> [:class]